source/hardware/SignalGP/utils/LinearFunctionsProgram.h

```diff
@@ -243,7 +248,18 @@ namespace sgp {
 
     // todo - various load from string functions (configurable?)
 
-    // todo - prints
+    // todo - support tabbing/levels for block type instructions
+    // todo - easier to read print without tags
+    template<typename HARDWARE_T, typename INST_PROPERTY_T>
+    void Print(std::ostream& out, const InstructionLibrary<HARDWARE_T, inst_t, INST_PROPERTY_T>& ilib) const{
+      for (size_t i = 0; i < GetSize(); ++i){
+        // Skip the last tag so we dont get an extra delimeter
+        std::copy(program[i].GetTags().begin(), program[i].GetTags().end() - 1, std::ostream_iterator<tag_t>(out, "\n"));
+        out << program[i].GetTags().back() << " Fn-"<< i+1 << "\n";
```

Comment:
why start function labels at 1 as opposed to 0?

Reply:
mostly cosmetic, Ill change it to 0-based